torch/distributions/wishart.py

```diff
@@ -0,0 +1,275 @@
+import math
+import warnings
+from numbers import Number
+from typing import Union
+
+import torch
+from torch.distributions import constraints
+from torch.distributions.exp_family import ExponentialFamily
+from torch.distributions.utils import lazy_property
+from torch.distributions.multivariate_normal import _precision_to_scale_tril
+
+
+_log_2 = math.log(2)
+
+
+def _mvdigamma(x: torch.Tensor, p: int) -> torch.Tensor:
+    assert x.gt((p - 1) / 2).all(), "Wrong domain for multivariate digamma function."
+    return torch.digamma(x.unsqueeze(-1) - torch.arange(p).div(2).expand(x.shape + (-1,))).sum(-1)
```

Comment:
shoudl torch.arange call also have `dtype = x.dtype, device = x.device`

Reply:
Thank you for your feedback. I added the arguments.